.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,156 @@
+            sytest-tag: bionic
+
+          - label: Py 3.6, PG 10, Monolith
+            sytest-tag: bionic
+            postgres: postgres
+
+          - label: Py 3.6, PG 10, Workers
+            sytest-tag: bionic
+            postgres: postgres
+            workers: workers
+
+          - label: Py 3.9, PG 13, Monolith
+            sytest-tag: testing
+            postgres: postgres
+
+          - label: Py 3.9, PG 13, Workers
+            sytest-tag: testing
+            postgres: postgres
+            workers: workers
+
+    container:
+      image: matrixdotorg/sytest-synapse:${{ matrix.sytest-tag }}
+      volumes:
+        # bootstrap.sh expects the sytest source available at /sytest.
+        # TODO Buildkite mounted /sytest as readonly. Can we do this on GHA? Do we need it?
+        - ${{ github.workspace }}/sytest:/sytest
+        # synapse_sytest.sh expects a synapse checkout at /src
+        - ${{ github.workspace }}/synapse:/src
+      env:
+        POSTGRES: ${{ matrix.postgres && 1 }}
+        WORKERS: ${{ matrix.workers && 1 }}
+        BLACKLIST: ${{ (matrix.workers && 'synapse-blacklist-with-workers') || 'sytest-blacklist' }}
+
+    steps:
+      - name: Checkout sytest
+        uses: actions/checkout@v2
+        with:
+          path: sytest
+
+      - name: Fetch corresponding synapse branch
+        shell: bash
+        run: |
+          BRANCH=${GITHUB_REF#refs/heads/}
+          (wget -O - https://github.com/matrix-org/synapse/archive/$BRANCH.tar.gz || wget -O - https://github.com/matrix-org/synapse/archive/develop.tar.gz) \
+            | tar -xz --strip-components=1 -C /src/
+
+      - name: Prepare blacklist file for running with workers
+        if: ${{ matrix.workers }}
+        run: cat /src/sytest-blacklist /src/.ci/worker-blacklist > /src/synapse-blacklist-with-workers
+
+      - name: Run sytest
+        run: |
+          echo POSTGRES=${POSTGRES:-<NOT SET>}
+          echo WORKERS=${WORKERS:-<NOT SET>}
+          echo BLACKLIST=${BLACKLIST:-<NOT SET>}
+          bash -xe /bootstrap.sh synapse
+
+      - name: Summarise results.tap
+        # Use always() to run this step even if previous ones failed.
+        if: ${{ always() }}
+        run:  /sytest/scripts/tap_to_gha.pl /logs/results.tap
+
+      - name: Upload SyTest logs
+        uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: Sytest Logs - ${{ job.status }} - (Synapse, ${{ matrix.label }})
+          path: |
+            /logs/results.tap
+            /logs/**/*.log*
+
+  dendrite:
+    runs-on: ubuntu-latest
+    name: "Dendrite: ${{ matrix.label }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - label: SQLite
+
+          - label: SQLite, full HTTP APIs
+            api: full-http
+
+          - label: Postgres
+            postgres: postgres
+
+          - label: Postgres, full HTTP APIs
+            postgres: postgres
+            api: full-http
```

Comment:
I suppose we don't have a clear idea of postgres versions here as there's no distribution label for `sytest-dendrite`?

Reply:
`sytest-dendrite:latest` looks to be based on Debian buster at the time of writing according to `docker/build.sh:16`. I was a bit hesitant to write in a PG version for a tag that could change under our feet. But we already do this for `sytest-synapse:testing`, based on `sytest:testing`, based on `debian:testing`.

Reply:
Good point, yeah. Actually, I suppose it might be a good idea to simply remove the version of `debian:testing` as well. I can't see us really remembering to ever update that version number...